src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -1039,6 +1039,14 @@ private async Task RefreshOutgoingTasksAsync()
                     var database = await completedTask;
                     await database.RefreshFeaturesAsync();
                 }
+                catch (OperationCanceledException)
+                {
+                    // database shutdown
+                }
+                catch (ObjectDisposedException)
+                {
+                    // database shutdown
+                }
```

Comment:
Are those additional catches directly related to the changes made here? If yes, then can you explain it a bit more?

Reply:
This is unrelated to the change. It should be there since RefreshFeaturesAsync can throw those exceptions. Those exceptions are already handled by all other callers.

Reply:
ok